controller/src/beerocks/master/tasks/client_steering_task.cpp

```diff
@@ -146,14 +147,25 @@ void client_steering_task::steer_sta()
         TASK_LOG(DEBUG) << "SLAVE " << sta_mac
                         << " has an active socket, sending BACKHAUL_ROAM_REQUEST";
         auto roam_request =
-            message_com::create_vs_message<beerocks_message::cACTION_CONTROL_BACKHAUL_ROAM_REQUEST>(
-                cmdu_tx, 0);
-        if (roam_request == nullptr) {
-            LOG(ERROR) << "Failed building message!";
+            cmdu_tx.create(0, ieee1905_1::eMessageType::BACKHAUL_STEERING_REQUEST_MESSAGE);
+        if (!roam_request) {
+            LOG(ERROR) << "Failed building BACKHAUL_STEERING_REQUEST_MESSAGE!";
             return;
         }
-        roam_request->params().bssid   = tlvf::mac_from_string(target_bssid);
-        roam_request->params().channel = database.get_node_channel(target_bssid);
+
+        auto bh_steer_req_tlv = cmdu_tx.addClass<wfa_map::tlvBackhaulSteeringRequest>();
+        if (!bh_steer_req_tlv) {
+            LOG(ERROR) << "Failed building addClass<wfa_map::tlvSteeringRequest!";
+            return;
+        }
+
+        bh_steer_req_tlv->backhaul_station_mac()  = tlvf::mac_from_string(sta_mac);
+        bh_steer_req_tlv->target_bssid()          = tlvf::mac_from_string(target_bssid);
+        bh_steer_req_tlv->target_channel_number() = database.get_node_channel(target_bssid);
```

Comment:
I don't really like that the result of `get_node_channel` is not checked here, but since it is not checked in the other places of the codebase, it is probably fine.